### PR TITLE
[1.21.x] TODOs and NBT cleanup for entities

### PR DESCRIPTION
**Pull request by robotgryphon**
- ~~Deprecates persistent data (use data attachments)~~
- Deprecates INBTSerializer for entities (data attachments/workarounds)
- Method renames for the 1.19 world -> level change

### CODE REVIEW

**Comment by Technici4n** [comment]
I believe that there are some uses cases for entity persistent data, for example marking item entities that should not be picked up by magnets, and other similar cases of mod compat.

I agree with the other changes.

**Comment by robotgryphon** [comment]
> I believe that there are some uses cases for entity persistent data, for example marking item entities that should not be picked up by magnets, and other similar cases of mod compat.
> 
> I agree with the other changes.

You'd rather keep persistent data when attachments exist, for stuff like magnet control? That feels like a capability or attachment thing.. persistent data is like using a sword when you have a scalpel nearby.

File: src/main/java/net/neoforged/neoforge/common/extensions/IEntityExtension.java [hunk 1]
```diff
@@ -135,24 +149,42 @@ default MobCategory getClassification(boolean forSpawnCount) {
...
      */
-    // TODO: rename in 1.19 to onRemovedFromLevel
-    void onRemovedFromWorld();
+    void onAddedToLevel();
+
+    /**
+     * @deprecated Use {@link #onAddedToLevel()} instead.
+     */
+    @Deprecated(forRemoval = true, since = "21.0")
+    default void onAddedToWorld() {
```
Comment (HenryLoenwind) [inline src/main/java/net/neoforged/neoforge/common/extensions/IEntityExtension.java]: Does this really need to stay around as baggage? NeoForge's still in the breaking-changes phase of 1.21.

**Comment by TelepathicGrunt** [comment]
Could we skip deprecating/removing persistentData? It is easier for mod compat without compile dependencies at the moment which Data Attachments doesn't fully replace very well at the moment

**Comment by HenryLoenwind** [comment]
> Could we skip deprecating/removing persistentData? It is easier for mod compat without compile dependencies at the moment which Data Attachments doesn't fully replace very well at the moment

Or could neo maybe provide a simple network-synced data attachment that's a `Map<String, String>` for mod interop? That way, the new system would be used, there'd be no compile-time dependency, and it'd be dead simple to use.

And as bonus, the javadoc could give two nice examples, `key="doNotMagnet", value=anything non-empty (preferably modId of the mod to set this)` to prevent an item from being attracted by magnets completely and "key="magnetReserved", value=any unique ID" to reserve an item for one specific magnet.

And no, we cannot prevent modders from stuffing their mod's whole save state into one of those Strings if they want.

**Review by sciwhiz12 — DISMISSED** [review]
Merge when ready.

**Review by sciwhiz12 — APPROVED** [review]
Merge when ready.

**Review by Technici4n — APPROVED** [review]

**Pull request merged**